### PR TITLE
shell.nix: minimize shell when not `isDevelopmentShell`

### PR DESCRIPTION
**Pull request by Profpatsch**
Remove development specific environment variables from derivation.

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -1,8 +1,9 @@
-{ pkgs ? import ./nix/nixpkgs.nix { enableMozillaOverlay = true; }
-, isDevelopmentShell ? true }:
+{ isDevelopmentShell ? true
```
Comment (curiousleo) [inline shell.nix]: Perhaps you could add a comment that explains the meaning or effect of setting / unsetting `isDevelopmentShell`.

**Comment by grahamc** [comment]
why?

File: shell.nix (hunk 1 above)
Comment (Profpatsch) [reply shell.nix]: Done, thanks.

**Comment by Profpatsch** [comment]
> why?

Unnecessary to pull in rust unstable on CI.

**Comment by Profpatsch** [comment]
I’ll just merge, we can revert should it make any problems, but the CI does have to set up less stuff if the shell is minimal.

**Pull request merged**